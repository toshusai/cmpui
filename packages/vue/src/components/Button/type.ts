export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
