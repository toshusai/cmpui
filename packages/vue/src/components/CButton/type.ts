export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
