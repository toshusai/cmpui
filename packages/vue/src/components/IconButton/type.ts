export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
