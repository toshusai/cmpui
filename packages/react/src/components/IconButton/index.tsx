import { forwardRef } from "react";

import { classNames } from "../../utils/classNames";

export type IconButton = {
  selected?: boolean;
  size?: "S" | "M";
} & JSX.IntrinsicElements["button"];

export const IconButton = forwardRef<HTMLButtonElement, IconButton>(
  function IconButton(
    { children, size = "M", selected, className, ...props },
    ref,
  ) {
    return (
      <button
        ref={ref}
        type="button"
        className={classNames("cmpui_icon-button__root", className)}
        aria-selected={selected}
        data-size={size}
        {...props}
      >
        {children}
      </button>
    );
  },
);
