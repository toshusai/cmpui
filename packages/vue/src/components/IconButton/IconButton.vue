<script setup lang="ts">
import { IconButtonProps } from "./type";

withDefaults(defineProps<IconButtonProps>(), {
  size: "M",
});
</script>

<template>
  <button
    class="cmpui_icon-button__root"
    :aria-selected="selected"
    :data-size="size"
    :disabled="disabled"
    :data-rounded="rounded"
    :data-border="border"
  >
    <slot></slot>
  </button>
</template>
