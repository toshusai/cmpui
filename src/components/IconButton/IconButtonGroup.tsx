import React, { forwardRef } from "react";
import { classNames } from "../../utils/classNames";

export const IconButtonGroup = forwardRef<
  HTMLDivElement,
  {
    children: React.ReactNode;
    className?: string;
  }
>((props, ref) => {
  return (
    <div
      ref={ref}
      className={classNames("cmpui_icon-button__group", props.className)}
    >
      {props.children}
    </div>
  );
});
