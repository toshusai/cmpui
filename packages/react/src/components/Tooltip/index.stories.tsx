import {
  IconArrowDown,
  IconArrowLeft,
  IconArrowRight,
  IconArrowUp,
  IconInfoCircle,
  IconPhoto,
  IconPointer,
  IconTypography,
} from "@tabler/icons-react";
import type { Meta, StoryObj } from "@storybook/react";

import { IconButton } from "../IconButton";

import { Tooltip } from ".";

const meta: Meta<typeof Tooltip> = {
  title: "Float/Tooltip",
  component: Tooltip,
};

export default meta;
type Story = StoryObj<typeof Tooltip>;

export const Basic: Story = {
  render: function RenderBasic() {
    return (
      <Tooltip content="Tooltip">
        <IconInfoCircle size={20} />
      </Tooltip>
    );
  },
};

export const Flip: Story = {
  render: function RenderBasic() {
    return (
      <Tooltip content="Tooltip">
        <IconInfoCircle
          size={20}
          style={{
            position: "absolute",
            top: "8px",
            right: "8px",
          }}
        />
      </Tooltip>
    );
  },
};

export const ForceShow: Story = {
  render: function RenderBasic() {
    return (
      <Tooltip content="Tooltip" forceShow>
        <IconInfoCircle size={20} />
      </Tooltip>
    );
  },
};

export const Placement: Story = {
  render: function RenderPlacement() {
    const sides = ["top", "right", "bottom", "left"] as const;
    const icons = {
      top: <IconArrowUp />,
      right: <IconArrowRight />,
      bottom: <IconArrowDown />,
      left: <IconArrowLeft />,
    };
    return (
      <div
        style={{
          display: "grid",
          gridTemplateColumns: "repeat(2, 150px)",
          gridTemplateRows: "repeat(2, 150px)",
        }}
      >
        {sides.map((side) => (
          <div
            key={side}
            style={{
              width: "100%",
              height: "100%",
              display: "flex",
              border: "1px solid var(--cmpui-border-color)",
            }}
          >
            <Tooltip content="Tooltip" side={side} key={side}>
              <IconButton
                style={{
                  margin: "auto",
                }}
              >
                {icons[side]}
              </IconButton>
            </Tooltip>
          </div>
        ))}
      </div>
    );
  },
};

export const Group: Story = {
  render: function RenderBasic() {
    return (
      <div className="cmpui_icon-button__group">
        <Tooltip content="Select">
          <IconButton>
            <IconPointer size={20} />
          </IconButton>
        </Tooltip>
        <Tooltip content="Text" side="bottom">
          <IconButton>
            <IconTypography size={20} />
          </IconButton>
        </Tooltip>
        <Tooltip content="Image">
          <IconButton>
            <IconPhoto size={20} />
          </IconButton>
        </Tooltip>
      </div>
    );
  },
};
