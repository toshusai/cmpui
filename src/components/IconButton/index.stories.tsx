import type { Meta, StoryObj } from "@storybook/react";
import { IconButton } from ".";
import { Icon } from "../../__stories__/Icon";
import { IconButtonGroup } from "./IconButtonGroup";

const meta: Meta<typeof IconButton> = {
  title: "Components/Actions/IconButton",
  component: IconButton,
};

export default meta;

type Story = StoryObj<typeof IconButton>;

const corner = (
  <svg
    width="6"
    height="6"
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    style={{
      position: "absolute",
      bottom: 4,
      right: 4,
      transform: "scaleX(-1)",
    }}
  >
    <title>corner</title>
    <path d="M0 24V0.00427246L24 24H0Z" fill="currentColor" />
  </svg>
);

export const Basic: Story = {
  render: () => {
    return (
      <div style={{ display: "flex", gap: 4 }}>
        <IconButton>
          <Icon filled name="arrow_selector_tool" style={{ fontSize: 20 }} />
        </IconButton>
        <IconButton selected>
          <Icon name="square" style={{ fontSize: 20 }} />
          {corner}
        </IconButton>
        <IconButton>
          <Icon name="title" style={{ fontSize: 20 }} />
        </IconButton>
      </div>
    );
  },
};

export const Selected: Story = {
  render: () => {
    return (
      <IconButton selected>
        <Icon name="star" style={{ fontSize: 20 }} />
      </IconButton>
    );
  },
};

export const Disabled: Story = {
  render: () => {
    return (
      <IconButton disabled>
        <Icon name="star" style={{ fontSize: 20 }} />
      </IconButton>
    );
  },
};

export const Small: Story = {
  render: () => {
    return (
      <IconButton size="S">
        <Icon name="star" style={{ fontSize: 16 }} />
      </IconButton>
    );
  },
};

export const Group: Story = {
  render: () => {
    return (
      <IconButtonGroup>
        <IconButton>
          <Icon filled name="align_horizontal_left" style={{ fontSize: 16 }} />
        </IconButton>
        <IconButton selected>
          <Icon name="align_horizontal_center" style={{ fontSize: 16 }} />
        </IconButton>
        <IconButton>
          <Icon name="align_horizontal_right" style={{ fontSize: 16 }} />
        </IconButton>
      </IconButtonGroup>
    );
  },
};
