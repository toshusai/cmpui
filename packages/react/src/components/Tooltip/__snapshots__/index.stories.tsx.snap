// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Float/Tooltip Basic smoke-test 1`] = `
<button type="button"
        class="cmpui_button"
        data-state="closed"
        data-variant="primary"
        data-size="M"
>
  Tooltip
</button>
`;

exports[`Float/Tooltip Multiple smoke-test 1`] = `
<div style="display: flex;">
  <button type="button"
          class="cmpui_icon-button__root"
          data-size="M"
          data-state="closed"
  >
    A
  </button>
  <button type="button"
          class="cmpui_icon-button__root"
          data-size="M"
          data-state="closed"
  >
    B
  </button>
  <button type="button"
          class="cmpui_icon-button__root"
          data-size="M"
          data-state="closed"
  >
    C
  </button>
</div>
`;

exports[`Float/Tooltip Presenter smoke-test 1`] = `
<div class="cmpui_tooltip__root">
  Hello, World!
</div>
`;
